gordon/interfaces.py

```diff
@@ -120,3 +120,46 @@ class IPublisherClient(IGenericPlugin):
         Args:
             event_msg (IEventMessage): Message ready to be sent to destination.
         """
+
+
+class IMetricRelay(Interface):
+    """Create and publish (push) metrics on demand."""
+
+    def incr(metric_name, value=1):
```

Comment:
For this and `timer`, I think having one method that does two things is of limited value and confusing. I also am worried that it is another layer of abstraction that has to be kept in step between the end user and the init method of the metric itself, which has shown to be a problem in Shumway.